File: Assignment_29/program2.c
////////////////////////////////////////////////////////////////////////
//
// Name :          program2.c
// Discription :   Count number small character from file
// Input :         String
// Output :        Integer
// Author :        Shivam Rajendra Kale
// Date :          12/06/2025
//
///////////////////////////////////////////////////////////////////////

/*
  Start
    Accept file name as input from user
        open that file in read mode than
         Count number small character from file
          Display count
  Stop    
*/

#include<stdio.h>
#include<stdlib.h>
#include<fcntl.h>
#include<io.h>
#include<string.h>

int CountSmall(char FName[])
{
    int fd = 0, iCnt = 0, iCount = 0, iRet = 0;

    char Buffer[10] = {'\0'};

    fd = open(FName,O_RDONLY);

    if(-1 == fd)
    {
        printf("Unaple to open file\n");
        return -1;
    }
    else
    {
        while(iRet = read(fd,Buffer,10))
        {
            for(iCnt = 0; iCnt < 10; iCnt++)
            {
                if((Buffer[iCnt] >= 'a') && (Buffer[iCnt] <= 'z'))
                {
                    iCount++;
                }
            }
            memset(Buffer,'\0',10);
        }

        close(fd);
    }

    return iCount;
}

int main()
{
    char Arr[50] = {'\0'};
    int iRet = 0;

    printf("Enter file name :\n");
    scanf("%s",Arr);

    iRet = CountSmall(Arr);

    printf("Number of small character are : %d",iRet);

    return 0;
}

File: Assignment_55/program2.java
//////////////////////////////////////////////////////////////////////////
//
// Name :          program2.java
// Discription :   use to count vowels character
// Input :         Integer
// Output :        Integer, String
// Author :        Shivam Rajendra Kale
// Date :          19/10/2025
//
//////////////////////////////////////////////////////////////////////////

import java.util.*;

class MyArray
{
    public int ArrayCountVowels(char Arr[])
    {
        int i = 0, Count = 0;
    
        for(i = 0; i < Arr.length; i++)
        {
           if(Arr[i] == 'A' || Arr[i] == 'E' || Arr[i] == 'I' || Arr[i] == 'O' || Arr[i] == 'U' || 
              Arr[i] == 'a' || Arr[i] == 'e' || Arr[i] == 'i' || Arr[i] == 'o' || Arr[i] == 'u')
           {
                Count++;
           }
        }
        return Count;
    }
}

class program2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        int i = 0, iRet = 0;

        System.out.println("Enter Array of Characters : ");
        String str = sobj.nextLine();

        char Arr[] = str.toCharArray();

        MyArray mobj = new MyArray();

        iRet = mobj.ArrayCountVowels(Arr);
        System.out.println("Vowels are : "+iRet);

        sobj.close();
    }

}
